Create API for `.ar*`, `aei`, `ar <reg>=<val>`

ret2libc opened the pull request:
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

<!-- Explain the **details** for making this change. Is a new feature implemented? What existing problem does the pull request solve? How does the pull request solve these issues? Please provide enough information so that others can review your pull request. -->

* Create API for `.ar*`, `aei`, `ar <reg>=<val>`
* Move rz_debug_reg_list from RzDebug to RzCore and directly use rz_cons_printf

**Test plan**

<!-- What steps should the reviewer take to test your pull request? Demonstrate the code is solid. Example: The exact commands you ran and their output, screenshots/videos. This is your time to re-check that everything works and that you covered all the edge cases -->

...

**Closing issues**

<!-- put "closes #XXXX" in your comment to auto-close the issue that your PR fixes (if any). -->

...

ret2libc committed:
Create API for `.ar*`, `aei`, `ar <reg>=<val>`

* Move rz_debug_reg_list from RzDebug to RzCore and directly use
  rz_cons_printf

wargio commented on librz/core/cdebug.c:
```diff
@@ -19,9 +20,222 @@ RZ_API bool rz_core_debug_step_one(RzCore *core, int times) {
+				}
+			}
+		} else {
+			value = rz_reg_get_value_big(dbg->reg, item, &valueBig);
```
value is never used after this assignment.

ret2libc replied:
this function is all copy-pasted code moved from rz_debug_reg_List :(

ret2libc replied:
didn't even look at it so far

wargio replied:
i guess you can just delete the line.

wargio commented on librz/core/agraph.c:
```diff
@@ -5,6 +5,7 @@
 #include <rz_util/rz_graph_drawable.h>
 #include <ctype.h>
 #include <limits.h>
+#include "private.h"
```
This header name feels wrong.

ret2libc replied:
suggestions? It is used also in other modules so far e.g. io_private.h , bin/i/private.h

wargio replied:
`io_private` seems much better

ret2libc committed:
rz_reg_get_list returns always the same list

rz_reg_get_list returns always the same list, so it shall not be freed
by the caller. Make rz_reg_get_list return a const RzList to make sure
it is not passed to rz_list_free.

ret2libc committed:
Avoid dead assignment to value

ret2libc committed:
Rename core/private.h to core/core_private.h

ret2libc committed:
Merge branch 'dev' into rzcorecmd-analysis

ret2libc committed:
Use reg->size / 8 as in the original code

ret2libc committed:
Select the right reg size when doing regs_to_flag

ret2libc committed:
Merge remote-tracking branch 'origin/dev' into rzcorecmd-analysis

ret2libc committed:
Merge remote-tracking branch 'origin/dev' into rzcorecmd-analysis

ret2libc committed:
Use rz_core_debug_regs2flags instead of .dr*

XVilka reviewed (changes requested):
I see almost everywhere `rz_core_debug_reg_list()` called with `core->dbg` as the first parameter. What about using `RzCore *core` then instead of `RzDebug *dbg`? It will make using corebind unnecessary.

ret2libc committed:
Make rz_core_debug_reg_list accept RzCore*

ret2libc commented:
> I see almost everywhere `rz_core_debug_reg_list()` called with `core->dbg` as the first parameter. What about using `RzCore *core` then instead of `RzDebug *dbg`? It will make using corebind unnecessary.

Good point, done.

ret2libc committed:
Merge branch 'dev' into rzcorecmd-analysis

XVilka commented on librz/core/cmd_analysis.c:
```diff
@@ -6107,31 +6142,8 @@ static void cmd_analysis_esil(RzCore *core, const char *input) {
 			rz_analysis_esil_free(esil);
 			core->analysis->esil = NULL;
 			break;
-		case 0: //lolololol
```
lolololol

ret2libc committed:
Merge branch 'dev' into rzcorecmd-analysis
